### PR TITLE
Properly add the dirname tag to all log entries for files discovered through wild card configuration.

### PR DESCRIPTION
**Pull request by remeh**
### What does this PR do?

The `searchFiles` method which is looking for files to tail in a directory is aware if a file has been discovered through a wild card patter but it was not creating the `File` object with this info, this flag was instead set in the regularly called method `FilesToTail`.
This behavior could lead to `File` objects not having the flag appropriately set during a certain amount of time, sending logs without the `dirname` tag.

Basically, this PR changes the `File` constructor in order to store this info as soon as the File object is created, it also removes this responsability to `FilesToTail`.

### Motivation

Properly add the `dirname` tag to log entries for files discovered through wild card configuration.

### CODE REVIEW

File: pkg/logs/input/file/file_provider.go [hunk 1]
```diff
@@ -150,8 +152,9 @@ func (p *Provider) searchFiles(pattern string, source *config.LogSource) ([]*Fil
 	sort.SliceStable(paths, func(i, j int) bool {
 		return filepath.Base(paths[i]) > filepath.Base(paths[j])
 	})
+	isWildcardPath := p.containsWildcard(pattern)
 	for _, path := range paths {
-		files = append(files, NewFile(path, source))
+		files = append(files, NewFile(path, source, isWildcardPath))
```
Comment (ajacquemot) [inline pkg/logs/input/file/file_provider.go]: I feel like you can just put `true` as we can only enter this method if the the path contains wildcard, see https://github.com/DataDog/datadog-agent/blob/c2636411379c03466615170f217b84dd87b82232/pkg/logs/input/file/file_provider.go#L123

**Review by ajacquemot — COMMENTED** [review]
Thanks for this PR, looks good  👍, I left a comment.

File: pkg/logs/input/file/file_provider.go (hunk 1 above)
Comment (remeh) [reply pkg/logs/input/file/file_provider.go]: Applying the change 👍

**Review by ajacquemot — DISMISSED** [review]

File: releasenotes/notes/logs-tag-dirname-at-creation-9da7c8f17a9b45bf.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    logs agent - tailed files discovered through a configuration entry with
```
Comment (KSerrania) [inline releasenotes/notes/logs-tag-dirname-at-creation-9da7c8f17a9b45bf.yaml]: Nit: I don't know we have a formal way to highlight changes with the Logs product, but I think we should have one (makes it easier to read in the changelog). We could use a convention similar to what we do for APM, like this:

```suggestion
    Logs: tailed files discovered through a configuration entry with
```

WDYT?

**Review by KSerrania — DISMISSED** [review]

**Review by KSerrania — APPROVED** [review]

**Review by ajacquemot — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]
